File: 1603-RunningSumOf1dArray/1603-RunningSumOf1dArray.js
// Last updated: 3/7/2026, 10:02:25 PM
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var runningSum = function(nums) {
    let arr = [];
    let sum = 0;

    for(let i=0; i<nums.length; i++){
        sum = 0;
        for(let j=i; j>=0; j--){
            sum += nums[j];
        }
        arr.push(sum);
    }
    return arr;
};

File: 1603-RunningSumOf1dArray/1603-RunningSumOf1dArray.cpp
// Last updated: 8/18/2026, 11:50:05 AM
class Solution {
public:
    vector<int> runningSum(vector<int>& nums) {
        int sum = 0;
        for(int i=0; i<nums.size(); i++){
            sum = sum + nums[i];
            nums[i] = sum;
        }
        return nums;
    }
};